contracts/okp4-objectarium/src/contract.rs

```diff
@@ -849,16 +848,38 @@ mod tests {
         instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
 
         let object = general_purpose::STANDARD.encode("already existing object");
-        let msg = ExecuteMsg::StoreObject {
-            data: Binary::from_base64(object.as_str()).unwrap(),
-            pin: true,
-            compression_algorithm: Some(CompressionAlgorithm::Passthrough),
-        };
-        execute(deps.as_mut(), mock_env(), info.clone(), msg.clone()).unwrap();
-        assert_eq!(
-            execute(deps.as_mut(), mock_env(), info, msg).err(),
-            Some(ContractError::Bucket(BucketError::ObjectAlreadyStored)),
+        execute(
+            deps.as_mut(),
+            mock_env(),
+            info.clone(),
+            ExecuteMsg::StoreObject {
+                data: Binary::from_base64(object.as_str()).unwrap(),
+                pin: false,
+                compression_algorithm: Some(CompressionAlgorithm::Passthrough),
+            },
+        )
+        .unwrap();
+
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            info.clone(),
+            ExecuteMsg::StoreObject {
+                data: Binary::from_base64(object.as_str()).unwrap(),
+                pin: true,
+                compression_algorithm: Some(CompressionAlgorithm::Passthrough),
+            },
         );
+
+        assert!(res.is_ok());
+        assert!(pins().has(
```

Comment:
Here, add an assertion test to ensure that when the stored object is retrievied, its `pin_count` equals 1 (the first storage action doesn't pin, but the second does). Currently, this is not working correctly, as the `pin_count` remains at 0.